R/packages.r

```diff
@@ -1,16 +1,31 @@
 #' Load a bunch of packages
 #' 
 #' @param ... List of packages
-#' @export
+#' @param verbose logical. Whether or not to announce installations and
+#' package startup messages.
 #' @examples
 #' \dontrun{
-#' packages('glmnet', 'caret')
-#' packages('glmnet caret')
-#' packages(list('glmnet', 'caret'), c('e1071', 'parallel multicore'), 'stringr')
+#' packages("glmnet")           # Can install from CRAN
+#' packages("robertzk/Ramd")    # Can install from GitHub
+#' packages("glmnet", "caret")  # can load multiple packages in one call
+#' packages("glmnet caret")     # Can just separate with a space
+#' packages("robertzk/Ramd", "hadley/dplyr", "peterhurford/batchman")  # Can load multiple
+#' packages("robertzk/Ramd", "glmnet")  # Can install from both CRAN and GitHub
+#' packages("robertzk/Ramd@@v0.3")      # Can load from versions
+#' packages("robertzk/Ramd@@0.3")
+#' packages("robertzk/Ramd@@fbe1aa0e36df289b27881d077635352e6debdbc1")  # Can load from refs
+#' packages(list("FeiYeYe/xgboost", subdir = "R-package"))              # Can load from subdirectories
 #' }
-packages <- function(...) {
-  split_packages <- function(string) strsplit(string, '[^a-zA-Z.$0-9_]+')[[1]]
-  pkgs <- unlist(lapply(unlist(list(...)), split_packages))
-  suppressPackageStartupMessages(sapply(pkgs, load_package))
+#' @export
+packages <- function(..., verbose = FALSE) {
+  split_packages <- function(string) strsplit(string, '[^a-zA-Z.$0-9_/@]+')[[1]]
+  if ("list" %in% vapply(list(...), class, character(1))) {
```

Comment:
`class` is not always length 1 -- this has a chance of erroring. I would instead use

```r
if ("list" %in% vapply(list(...), function(x) class(x)[1L], character(1))) 
```

The `else` route here in general though seems to be no longer necessary. Can we simply use `pkgs <- list(...)` going forward?